src/common/io_util.cc

```diff
@@ -312,6 +314,53 @@ int AeWait(int fd, int mask, int timeout) {
   }
 }
 
+bool MatchListeningIP(std::vector<std::string> &binds, const std::string &ip) {
+  if (std::find(binds.begin(), binds.end(), ip) != binds.end()) {
+    return true;
+  }
+
+  // If binds contains 0.0.0.0, we should resolve ip addresses and check it
+  if (std::find(binds.begin(), binds.end(), "0.0.0.0") != binds.end()) {
+    auto local_ip_addresses = GetLocalIPAddresses();
+    return std::find(local_ip_addresses.begin(), local_ip_addresses.end(), ip) != local_ip_addresses.end();
+  }
+  return false;
+}
+
+// Get all local IP addresses.
+std::vector<std::string> GetLocalIPAddresses() {
+  std::vector<std::string> ip_addresses;
+  ifaddrs *if_addr_struct = nullptr;
+  // Use unique_ptr for if_addr_struct to avoid manually free
+  std::unique_ptr<ifaddrs, decltype(&freeifaddrs)> ifaddrs_ptr(nullptr, &freeifaddrs);
+  if (getifaddrs(&if_addr_struct) == -1) {
+    return ip_addresses;
+  }
+  ifaddrs_ptr.reset(if_addr_struct);
+
+  for (ifaddrs *ifa = if_addr_struct; ifa; ifa = ifa->ifa_next) {
+    if (!ifa->ifa_addr) {
+      continue;
+    }
+    void *tmp_addr_ptr = nullptr;
+    if (ifa->ifa_addr->sa_family == AF_INET) {
+      // check it is IP4 and not a loopback address
```

Comment:
The comment states that there is a check that it's not a loopback address, but I don't see any checks. Is everything OK here?
The same for the `else` branch.

Reply:
I rechecked this code, I didn't check the local loopback address, I think it's ok, so I should delete the extra comment after that, right? or should I check if it's the local loopback address?

Reply:
Yes, the comment should be removed if we didn't check that. It looks no necessary to check if it's a loopback address.

Reply:
The comment should be what the code does, otherwise it must be `TODO` or `FIXME`.